internal/rpcserver/interface.go

```diff
@@ -173,6 +170,13 @@ type SyncManager interface {
 	// insertion into the memory pool.
 	ProcessTransaction(tx *dcrutil.Tx, allowOrphans bool, rateLimit bool,
 		allowHighFees bool, tag mempool.Tag) ([]*dcrutil.Tx, error)
+
+	// ExistsAddress returns whether or not an address has been seen before.
+	ExistsAddress(addr dcrutil.Address) (bool, error)
```

Comment:
While I definitely agree with and like the approach of just exposing these and allowing the adaptor code to handle checking the implementation details of optionality, I'm not really sure that `SyncMgr` is really the right interface for it.

I realize the `ExistsAddrIndex` is exposed via the `SyncMgr` currently and that is likely why it was done there in keeping with the spirit of that.

This isn't something I would hold up the PR over, but I think it goes along with what I mentioned in the development channel that the interfaces really need to be pared down and reworked a bit.

I would think that all the RPC server cares about here is the ability to perform address existence checks.  That, to me, implies there would be an interface like:

```Go
type ExistsAddresser interface {
	ExistsAddress(addr dcrutil.Address) (bool, error)
	ExistsAddresses(addrs []dcrutil.Address) ([]bool, error)
}
```

Reply:
I made this change.

Reply:
Good call, moving to smaller interfaces like this will definitely be more maintainable